u8timeseries/tests/test_timeseries.py

```diff
@@ -428,3 +428,26 @@ def test_fill_missing_dates(self):
         self.assertTrue(series_test.start_time() == range_[0])
         self.assertTrue(series_test.end_time() == range_[-1])
         self.assertTrue(math.isnan(series_test.pd_series().get('20130105')))
+
+    def test_upsample_timeseries(self):
+        times = pd.date_range('20130101', '20130110')
+        pd_series = pd.Series(range(10), index=times)
+        timeseries = TimeSeries(pd_series)
+
+        resampled_timeseries = timeseries.upsample('H')
+        self.assertEqual(resampled_timeseries.freq_str(), 'H')
+        self.assertEqual(resampled_timeseries.pd_series().at[pd.Timestamp('20130101020000')], 0)
```

Comment:
You should be able to index directly into the `TimeSeries`. Something like `resampled_timeseries[pd.Timestamp('20130101020000')]`